lib/abci/handlers/query/verifyChainlockQueryHandlerFactory.js

```diff
@@ -0,0 +1,41 @@
+const { ChainLock } = require('@dashevo/dashcore-lib');
+
+const {
+  abci: {
+    ResponseQuery,
+  },
+} = require('abci/types');
+
+const InvalidArgumentAbciError = require('../../errors/InvalidArgumentAbciError');
+
+/**
+ *
+ * @param {SimplifiedMasternodeList} simplifiedMasternodeList
+ * @param {decodeChainLock} decodeChainLock
+ * @return {verifyChainlockQueryHandler}
+ */
+function verifyChainlockQueryHandlerFactory(simplifiedMasternodeList, decodeChainLock) {
+  /**
+   * @typedef identityQueryHandler
+   * @param {Object} params
+   * @param {Object} data
+   * @param {Buffer} data.chainLock
+   * @return {Promise<ResponseQuery>}
+   */
+  async function verifyChainlockQueryHandler(params, data) {
+    const chainlock = decodeChainLock(data.chainLock);
+
+    if (!chainlock.verify(simplifiedMasternodeList.getStore())) {
+      throw new InvalidArgumentAbciError(
+        `Signature invalid for chainlock
```

Comment:
Let's pass chainlock.toJSON() to the data param (second param of `InvalidArgumentAbciError`) instead of message.